client/src/auth/index.jsx

```diff
@@ -11,10 +10,13 @@ const PrivateRoute = ({ component: Component, ...rest }) => {
         return isAuth() ? (
           <Component {...props} />
         ) : (
-          <Redirect
-            to={{ pathname: '/login', state: { from: props.location } }}
-          />
-        );
+            <>
+              {rest.clearUserInfo()}
```

Comment:
@ashatat, this way, I got an error, 'can't update state while rendering.
 So how can I clear ```UserInfo``` and reset ``isLogin``` to false in case a user is not authenticated?